### PR TITLE
Various fixes for the item kit feature.

### LINKED ISSUE #1950: "Item Kit Items" printing on receipts regardless of "Kit Only" print option enabled.
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.20
- OSPOS git commit hash is: 4f5ad57
- PHP version is: 7.0.28
- MySQL or MariaDB version is: 5.7.17
- OS and version is: Ubuntu 16.04.1
- WebServer is: Apache 2.4.18
- Selected language is: English
- (If applicable) Docker installation: 
- (If applicable) Installation package for the LAMP/LEMP stack is: 

### Issue / Bug  / Question / New Feature

Please write your issue here. If a bug, please make sure to provide as much information as possible including configuration settings (e.g. Decimals set, Tax mode), language and steps to reproduce the bug.

1) I have the following options selected for a test Item Kit I created:

**Price Option:** Kit Only
**Print Option:** Priced Only

2) The Item kit is comprised of 2 Item Kit Items:

First Test Item
Second Test Item

3) The item kit also has the "Kit Item" field populated with the following item name "Oakley Test Item Kit"

When pulling up the item kit in the sales register, "Oakley Test Item Kit" appears with its correct pricing and "First Test Item" and "Second Test Item" appear with $0 for their pricing, as expected due to the 'Kit Only' price option.

However, when completing the sale and viewing the receipt, all 3 line items appear on the receipt instead of just the item kit, regardless of whether the 'Priced Only' or 'Kit Only' print option is selected prior to making the sale.  I have tried creating multiple different kit items with every possible combination of Price and Print options to no avail.  From the documentation online, the expected behavior would be to see only the Item Kit listed on the receipt, however, these doesn't appear to be the case.  Can you please recreate the issue on your end to confirm if my findings are accurate?

Let me know if you need screenshots or database entry info in order to further clarify the issue.

Thanks!

### PR DESCRIPTION
**Pull request by SteveIreland**
This fixes #1950 and resolves a couple of other problems I stumbled across in the area of  kit maintenance.

### CODE REVIEW

**Comment by mckaygerhard** [comment]
hi!, this fixeds will be soon in a release for 3.2.X ?

**Comment by SteveIreland** [comment]
Generally after a release we give it time to simmer and allow everyone a chance to find and report issues.  Unfortunately a lot of testing doesn't occur until after a new release.  So I wouldn't expect a 3.2.x release soon.  But depending on the severity of the issues they might put a rush on it.

**Comment by mckaygerhard** [comment]
umm i still have the subsequent sql repeated querys not resolved, i hope some one give me a hand due i cannot found the problem, that problem was many times before 3.2.0 release

File: application/controllers/Item_kits.php [hunk 1]
```diff
@@ -50,6 +58,8 @@ public function search()
 		$sort   = $this->input->get('sort');
 		$order  = $this->input->get('order');
 
+		error_log('>>>search started');
```
Comment (daN4cat) [inline application/controllers/Item_kits.php]: Debug code left behind....

**Pull request merged**